var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -143,6 +151,35 @@ class Esmf(MakefilePackage):
     # https://github.com/spack/spack/issues/35957
     patch("esmf_cpp_info.patch")
 
+    @when("+python")
+    def patch(self):
+        # The pyproject.toml file uses a dynamically generated version from git
+        # Howeer, this results in a version of 0.0.0 and a mismatch with the loaded version
+        # so this hardcodes it to match the library's version
+        filter_file(
+            """dynamic = \\[ "version" \\]""",
+            f"""version = "{self.version}" """,
+            os.path.join("src/addon/esmpy/pyproject.toml"),
+        )
+
+    def setup_dependent_run_environment(self, env, dependent_spec):
```

Comment:
Is there a way to define there for both build systems at once? I don't remember...

Reply:
Also, which one gets used when `+python`? Does the PythonExtension one supersede the MakefilePackage one?

Reply:
>Is there a way to define these for both build systems at once?

It was not clear to me that the Package propagated down to the instance of the `PipBuilder`.

>Also, which one gets used when +python? 

For the makefile portion, it remains the one defined in the main class. For the python pip invocation, the one in the PipBuilder class.

>Does the PythonExtension one supersede the MakefilePackage one?

My understanding is that `PythonExtension` is just a [meta-data helper mixin](https://spack.readthedocs.io/en/latest/packaging_guide.html#mixin-base-classes) that coordinates ensuring everything gets [loaded into the view correctly](https://spack.readthedocs.io/en/latest/spack.build_systems.html#spack.build_systems.python.PythonExtension) and in sync. 

Because the main class doesn't inherit from a PythonPackage, it doesn't invoke the builder class automatically (as I understand it and from testing). Thus, I can manually invoke the PipBuilder after the makefile install.

A makefile + python build seems uncommon in Spack packages and the closest example I could find was 
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-pybind11/package.py
which is what I modelled this after.

This is my pieced together understanding; I'm very open to being wrong. However, this does work as expected.

Reply:
Thanks. To clarify, my concern is about which of the `setup_dependent_run_environment()` and `setup_run_environment()` definitions get used. Ideally it would be nice to avoid having to redundantly update both definitions if/when changes are needed.

Reply:
Oh, I see what you're saying. Yes I agree. Did a bunch of testing and my assumptions about what was needed was wrong -- I have removed the redundent calls.